packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -249,10 +257,27 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
             // ODSP document ids are random guids (different per session)
             // fix the branch name in attributes
             // this prevents issues when generating summaries
-            const documentAttributes: api.IDocumentAttributes = JSON.parse(fromBase64ToUtf8(blob.content));
-            documentAttributes.branch = this.documentId;
+            if (blob.encoding === "base64") {
+                const documentAttributes: api.IDocumentAttributes = JSON.parse(fromBase64ToUtf8(blob.content));
+                documentAttributes.branch = this.documentId;
+
+                if (outputFormat === "base64") {
```

Comment:
Couple things:
First, we should figure out how remove this code to patch branch info. Matt would know better here, but we are removing all the code related to branching.
Second, even if you need to modify blob, there is no requirements that you encode it in exactly same encoding - you can always put back a string, not base64 content, as long we you change blob.encoding to reflect it.

Reply:
Re: the first point, the branching removal stuff just went in, so waiting a version before removing the parts in the protocol.